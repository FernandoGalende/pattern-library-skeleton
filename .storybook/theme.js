import { create } from '@storybook/theming/create';

const theme = create({
  base: 'light',
  //brandTitle: 'Pattern library skeleton',
  brandTitle: '#zapatillasFromMarsðŸ‘ŸðŸš€',
  brandUrl: 'https://github.com/klaufel/pattern-library-skeleton',
  //brandImage: 'your-image-url',
});

export default theme;
