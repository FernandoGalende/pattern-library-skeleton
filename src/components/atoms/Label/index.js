import PropTypes from 'prop-types'
import styled from 'styled-components'

const Label = styled.label`
  display: block;
  font-size: ${props => props.theme.typography.body.normal.fontSize};
  line-height: ${props => props.theme.typography.body.normal.lineHeight};
  padding-bottom: ${props => props.theme.spacing.xxs};
`

Label.propTypes = {
  /** This Boolean attribute prevents the user from interacting with the button */
  disabled: PropTypes.bool
}

export default Label
