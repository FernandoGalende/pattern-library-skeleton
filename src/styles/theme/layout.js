const layout = {
  grid: {
    width: '1624px',
    columns: 12,
    gutter: {
      xs: '8px',
      sm: '16px',
      md: '24px',
      lg: '24px',
      xl: '24px'
    },
    margin: {
      xs: '16px',
      sm: '24px',
      md: '32px',
      lg: '32px',
      xl: '32px'
    }
  },
  breakpoints: {
    xs: '576px', // xs: The number of columns to span on small devices (≥576px)
    sm: '768px', // sm: The number of columns to span on medium devices (≥768px)
    md: '992px', // md: The number of columns to span on large devices (≥992px)
    lg: '1200px', // lg: The number of columns to span on extra large devices (≥1200px)
    xl: '1560px' // xl: The number of columns to span on extra large devices (≥1560px)
  }
}

export default layout
