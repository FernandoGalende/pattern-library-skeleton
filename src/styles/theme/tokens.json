{
  "colors": {
    "alertWarning": "#ffa90d",
    "alertError": "#e5363a",
    "alertSuccess": "#109c42",
    "neutral00": "#ffffff",
    "neutral05": "#f2f2f2",
    "neutral20": "#cccccc",
    "neutral40": "#999999",
    "neutral80": "#333333",
    "primaryDark": "#1d698f",
    "primaryMain": "#b8255a",
    "primaryLight": "#56a4c9"
  },
  "typography": {
    "headingBig": {
      "fontFamily": "Open Sans",
      "fontSize": "28px",
      "lineHeight": "32px",
      "fontWeight": 700
    },
    "headingNormal": {
      "fontFamily": "Open Sans",
      "fontSize": "24px",
      "lineHeight": "28px",
      "fontWeight": 700
    },
    "headingSmall": {
      "fontFamily": "Open Sans",
      "fontSize": "18px",
      "lineHeight": "21px",
      "fontWeight": 700
    },
    "bodyNormal": {
      "fontFamily": "Open Sans",
      "fontSize": "14px",
      "lineHeight": "16px",
      "fontWeight": 400
    }
  },
  "spacing": {
    "xxl": "48px",
    "xl": "40px",
    "lg": "32px",
    "md": "24px",
    "sm": "16px",
    "xs": "8px",
    "xxs": "4px"
  }
}