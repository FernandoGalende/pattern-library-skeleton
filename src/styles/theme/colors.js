import tokens from './tokens';

const colors = {
  primary: {
    light: tokens.colors.primaryLight,
    main: tokens.colors.primaryMain,
    dark: tokens.colors.primaryDark,
  },
  neutral: {
    percent00: tokens.colors.neutral00,
    percent05: tokens.colors.neutral05,
    percent20: tokens.colors.neutral20,
    percent40: tokens.colors.neutral40,
    percent80: tokens.colors.neutral80,
  },
  alert: {
    success: tokens.colors.alertSuccess,
    warning: tokens.colors.alertWarning,
    error: tokens.colors.alertError,
  },
};

export default colors;
