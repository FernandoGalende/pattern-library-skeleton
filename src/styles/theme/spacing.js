import tokens from './tokens';

const spacing = {
  xxs: tokens.spacing.xxs,
  xs: tokens.spacing.xs,
  sm: tokens.spacing.sm,
  md: tokens.spacing.md,
  lg: tokens.spacing.lg,
  xl: tokens.spacing.xl,
  xxl: tokens.spacing.xxl,
};

export default spacing;
