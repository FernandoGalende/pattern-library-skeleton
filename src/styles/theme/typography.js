import tokens from './tokens';

const typography = {
  global: {
    family: "'Open Sans'",
  },
  heading: {
    big: {
      fontSize: tokens.typography.headingBig.fontSize,
      lineHeight: tokens.typography.headingBig.lineHeight,
      fontWeight: tokens.typography.headingBig.fontWeight,
    },
    normal: {
      fontSize: tokens.typography.headingNormal.fontSize,
      lineHeight: tokens.typography.headingNormal.lineHeight,
      fontWeight: tokens.typography.headingNormal.fontWeight,
    },
    small: {
      fontSize: tokens.typography.headingSmall.fontSize,
      lineHeight: tokens.typography.headingSmall.lineHeight,
      fontWeight: tokens.typography.headingSmall.fontWeight,
    },
  },
  body: {
    normal: {
      fontSize: tokens.typography.bodyNormal.fontSize,
      lineHeight: tokens.typography.bodyNormal.lineHeight,
    },
  },
};

export default typography;
