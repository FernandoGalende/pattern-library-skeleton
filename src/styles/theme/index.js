import colors from './colors'
import typography from './typography'
import spacing from './spacing'
import layout from './layout'
import shadows from './shadows'

const theme = {
  colors,
  typography,
  spacing,
  layout,
  shadows
}

export default theme
