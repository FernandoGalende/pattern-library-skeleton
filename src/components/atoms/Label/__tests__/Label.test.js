import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from '../../../../styles/theme';
import Label from '..';

describe('Label', () => {
  test('it works default', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Label>Default label</Label>
      </ThemeProvider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
