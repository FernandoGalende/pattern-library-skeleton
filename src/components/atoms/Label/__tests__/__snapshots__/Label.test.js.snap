// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label it works default 1`] = `
.c0 {
  display: block;
  font-size: 14px;
  line-height: 16px;
  padding-bottom: 4px;
}

<label
  class="c0"
>
  Default label
</label>
`;
